zulipterminal/model.py

```diff
@@ -821,12 +821,41 @@ def _get_stream_by_id(self, streams: List[StreamData], stream_id: int
                 return stream
         raise RuntimeError("Invalid stream id.")
 
+    def _unsubscribe_to_streams(self,
+                                stream_info_list: List[Subscription]) -> None:
+        is_pinned_stream_removed = False
+        is_unpinned_stream_removed = False
+        for stream_info in stream_info_list:
+            stream_id = stream_info['stream_id']
+            if self.is_user_in_stream(stream_id, self.pinned_streams):
+                stream = self._get_stream_by_id(self.pinned_streams,
+                                                stream_id)
+                self.pinned_streams.remove(stream)
+                is_pinned_stream_removed = True
+            else:
+                stream = self._get_stream_by_id(self.unpinned_streams,
+                                                stream_id)
+                self.unpinned_streams.remove(stream)
+                is_unpinned_stream_removed = True
+                sort_streams(self.unpinned_streams)
+            if not self.stream_dict[stream_id]:
+                self.muted_streams.remove(stream_id)
```

Comment:
Could you explain what this is intended to do? This won't avoid a `KeyError`.

Reply:
Updated the logic.